src/Compilers/VisualBasic/Test/Emit/Emit/CompilationEmitTests.vb

```diff
@@ -297,12 +297,246 @@ End Class
+                        If expectEmptyOrThrowNull Then
+                            AssertEx.Equal(throwNull, il)
+                        Else
+                            AssertEx.NotEqual(throwNull, il)
+                        End If
+                    End If
+                Next
+            End Using
+        End Sub
+
+        <Fact>
+        Sub EmitMetadataOnly_NoPdbs()
+            Dim comp = CreateCompilation("", references:={MscorlibRef},
+                            options:=TestOptions.DebugDll.WithDeterministic(True))
+
+            Using output As New MemoryStream()
+                Using pdbOutput = New MemoryStream()
+                    Assert.Throws(Of ArgumentException)(Function() comp.Emit(output, pdbOutput,
+                                        options:=EmitOptions.Default.WithEmitMetadataOnly(True)))
+                End Using
+            End Using
+        End Sub
+
+        <Fact>
+        Sub EmitMetadataOnly_NoMetadataPeStream()
+            Dim comp = CreateCompilation("", references:={MscorlibRef},
+                            options:=TestOptions.DebugDll.WithDeterministic(True))
+
+            Using output As New MemoryStream()
+                Using metadataPeOutput As New MemoryStream()
+                    Assert.Throws(Of ArgumentException)(Function() comp.Emit(output, metadataPeStream:=metadataPeOutput,
+                                        options:=EmitOptions.Default.WithEmitMetadataOnly(True)))
+                End Using
+            End Using
+        End Sub
+
+        <Fact>
+        Sub EmitMetadata()
+            Dim source =
+"Public MustInherit Class PublicClass
+    Public Sub PublicMethod
+        System.Console.Write(""Hello"")
+    End Sub
+End Class "
+            Dim comp = CreateCompilation(source, references:={MscorlibRef},
+                            options:=TestOptions.DebugDll.WithDeterministic(True))
+
+            Using output As New MemoryStream()
+                Using pdbOutput As New MemoryStream()
+                    Using metadataOutput As New MemoryStream()
+                        Dim result = comp.Emit(output, pdbOutput, metadataPeStream:=metadataOutput)
+                        Assert.True(result.Success)
+                        Assert.NotEqual(0, output.Position)
+                        Assert.NotEqual(0, pdbOutput.Position)
+                        Assert.NotEqual(0, metadataOutput.Position)
+                    End Using
+                End Using
+            End Using
+
+            Dim peImage = comp.EmitToArray()
+            VerifyMethodBodies(peImage, expectEmptyOrThrowNull:=False)
+        End Sub
+
+        <WorkItem(4344, "DevDiv_Projects/Roslyn")>
+                                         <Fact>
         Public Sub Bug4344()
             Dim compilation = CompilationUtils.CreateCompilationWithMscorlibAndVBRuntime(
 <compilation>
-    <file name="a.vb">
+                                                 <file name="a.vb">
```

Comment:
Here and throughout the file there are a lot of unexpected alignment changes. Please revert.